lib/ipam/ipam.go

```diff
@@ -195,9 +195,52 @@ func (c ipamClient) getBlockFromAffinity(ctx context.Context, aff *model.KVPair)
 	return b, nil
 }
 
-func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, pools []net.IPNet, version ipVersion, host string) ([]net.IP, error) {
+func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, requestedPools []net.IPNet, version ipVersion, host string) ([]net.IP, error) {
+	// Start by sanitizing the requestedPools.
+	pools := []net.IPNet{}
 
-	// Start by trying to assign from one of the host-affine blocks.  We
+	// Get all the configured pools.
+	enabledPools, err := c.pools.GetEnabledPools(version.Number)
+	if err != nil {
+		log.WithError(err).Errorf("Error reading configured pools")
+		return nil, err
+	}
+	log.Debugf("enabled IPPools: %v", enabledPools)
+
+	// Only add requested pools if they are enabled.
+	if len(requestedPools) > 0 {
+		log.Debugf("requested IPPools: %v", requestedPools)
+		for _, p := range enabledPools {
+			if isPoolInRequestedPools(p, requestedPools) {
+				pools = append(pools, p)
+			}
+		}
+	} else {
+		// If no pools were requested, default to all enabled pools.
+		pools = enabledPools
+	}
+	log.Debugf("Finding an unclaimed block from pools: %v", pools)
```

Comment:
This log is wrong now - we're not finding an unclaimed block yet (we might use an already claimed one)